poetry/console/commands/shell.py

```diff
@@ -0,0 +1,52 @@
+from os import environ
+from distutils.util import strtobool
+
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        # Check if it's already activated or doesn't exist and won't be created
+        if strtobool(environ.get("POETRY_ACTIVE", "0")) or not self.venv.is_venv():
```

Comment:
hm... Any particular reason to use outside library for this comparison?

Sorry if I'm being nitpicky, but from my POV it is unnescarry complication: now a person reading the code has to think about how this function behaves for example when `POETRY_ACTIVE` would be equal to  '3'.

What do you think about plain and simple:
```
venv_in_env = environ.get("POETRY_ACTIVE")
if venv_in_env or not self.venv.is_venv():
...
```

Reply:
What you suggested is going to evaluate any non-empty string as true, even `"0"` and `"false"`. What I did is a bit more robust. I only set `"1"`, but if someone sets it to `"true"` or other true-like value, manually or somewhere else in the code, it will work properly.

Reply:
Oh, and it's a module from the standard library, available in python 2.7 as well. I see no harm in that :)

Reply:
Yep, I'm getting tired :) I edited my answer actually to show what I had in mind (now it evaluates only "1" as true) but actually what you wrote convinces me 👍